### DISCUSSION TITLE
Can't get cell volumes in pipeflow case

### DISCUSSION THREAD

**Discussion by mojtabaAmiraslanpour** [UvaCsl/HemoCell#43] [General discussion]
Hello all,

I am using the same piece of code that has been used in the "oneCellShear" case to extract the average cells volume in the "pipeflow" case, but with no luck.

```
    // Get undeformed cell values
  CellInformationFunctionals::calculateCellVolume(&hemocell);
  CellInformationFunctionals::calculateCellArea(&hemocell);
  T volume_eq = (CellInformationFunctionals::info_per_cell[0].volume)/pow(1e-6/param::dx,3);
  T surface_eq = (CellInformationFunctionals::info_per_cell[0].area)/pow(1e-6/param::dx,2);
  pcout << "\t Surface: " << surface_eq << "  Volume: " << volume_eq << endl;
  
  while (hemocell.iter < tmax ) {
    hemocell.iterate();

    // Fill up the static info structure with desired data
    CellInformationFunctionals::calculateCellVolume(&hemocell);
    CellInformationFunctionals::calculateCellArea(&hemocell);

    T volume = (CellInformationFunctionals::info_per_cell[0].volume)/pow(1e-6/param::dx,3);
    T surface = (CellInformationFunctionals::info_per_cell[0].area)/pow(1e-6/param::dx,2);

...
```

I get zero for `surface `and `volume`.

Can you please tell me what I am doing wrong?

**Comment by gzavo** [comment]
Hi,

A quick guess: the cell information functionals were written for a single core application (one cell shear). Probably you want to use it in a distributed run with many cells.

**Reply by mojtabaAmiraslanpour** [reply]
Thanks, Gabor, is there any functionality to do this in HemoCell?
Any examples done it maybe?

**Reply by mojtabaAmiraslanpour** [reply]
Bad news, even when I am running the case with single-core I get 0 values.

**Reply by mojtabaAmiraslanpour** [reply]
Ok just found out that single-core also means having one cell in the domain. That works yup!
Now I want to know if it is straightforward to do this for multiple cells or not. does it need to deal with parallel computing stuff?
